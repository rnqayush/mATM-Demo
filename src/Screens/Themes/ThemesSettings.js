import React, { useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import { Button } from '@idfc/ccl-mobile';
import useThemes from '@optimus/themes/useThemes';
import Labels from 'translations/settingsScreen/labels';
import patternsHOC from 'helpers/patterns/patternsHOC';
import { ThemeContainer, StyledSafeAreaView, LabelText } from './ThemesSettings.style';
import ThemesOptions from './ThemesOptions';

const ThemesSettings = () => {
  const navigation = useNavigation();
  const { selectedThemeName, setSelectedThemeName } = useThemes();
  const [selectedThemeOption, setSelectedThemeOption] = useState(selectedThemeName);

  const onThemeChange = selectedItem => {
    setSelectedThemeOption(selectedItem);
  };

  const onApplyTheme = () => {
    setSelectedThemeName(selectedThemeOption);
    navigation.goBack();
  };

  return (
    <StyledSafeAreaView>
      <ThemeContainer>
        <LabelText fontSize={16} lineHeight={24} color="grey500" align="center">
          {Labels.SWITCH_THEME_LABEL}
        </LabelText>
        <ThemesOptions selectedThemeOption={selectedThemeOption} onThemeChange={onThemeChange} />
      </ThemeContainer>
      <Button testID="themeChange" title={Labels.SAVE} onPress={onApplyTheme} />
    </StyledSafeAreaView>
  );
};

export default patternsHOC({ Component: ThemesSettings });
