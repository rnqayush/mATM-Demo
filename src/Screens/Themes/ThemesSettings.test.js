import React from 'react';
import { render, fireEvent } from 'test/test-utils';
import ThemesSettings from './ThemesSettings';

const mockGoBack = jest.fn();

const mockedSetSelectedThemeName = jest.fn();

jest.mock('@optimus/themes/useThemes', () => () => ({
  setSelectedThemeName: mockedSetSelectedThemeName,
  selectedThemeName: 'LIGHT',
}));

jest.mock('react-native-appearance', () => {
  const ReactNative = jest.requireActual('react-native');
  return {
    Appearance: {
      getColorScheme: jest.fn().mockReturnValue('light'),
      addChangeListener: jest.fn().mockReturnValue({ remove: jest.fn() }),
    },
    AppearanceProvider: ({ children }) => <ReactNative.View>{children}</ReactNative.View>,
  };
});

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    goBack: () => mockGoBack(),
  }),
}));

jest.mock('@idfc/ccl-mobile', () => {
  const RealModule = jest.requireActual('test/ccl-mobile.mock');
  const ReactNative = jest.requireActual('react-native');
  const RenderOnlyProps = props => <ReactNative.View {...props} />;
  return {
    ...RealModule,
    RadioButton: RenderOnlyProps,
    Text: RenderOnlyProps,
    Button: RenderOnlyProps,
  };
});

jest.mock('@optimus/core-feature-toggle');

describe('ThemeSettings screen', () => {
  it('should render component', () => {
    const tree = render(<ThemesSettings />);
    expect(tree).toMatchSnapshot();
  });

  it('should apply dark theme when dark mode radio button is clicked and save button is pressed', () => {
    const container = render(<ThemesSettings />);
    const { getByTestId } = container;
    fireEvent(getByTestId('1-DARK'), 'onChange');
    fireEvent(getByTestId('themeChange'), 'onPress');

    expect(mockedSetSelectedThemeName).toHaveBeenCalledTimes(1);
    expect(mockedSetSelectedThemeName).toHaveBeenCalledWith('DARK');

    fireEvent(getByTestId('2-SYSTEM'), 'onChange');
    fireEvent(getByTestId('themeChange'), 'onPress');

    expect(mockedSetSelectedThemeName).toHaveBeenCalledWith('SYSTEM');
  });
});
