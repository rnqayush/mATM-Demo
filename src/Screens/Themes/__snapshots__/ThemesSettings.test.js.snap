// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSettings screen should render component 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "backgroundColor": "#f0f0f0",
      },
      Object {
        "flex": 1,
        "padding": 0,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "SvgMock",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "resizeMode": "cover",
        },
      ]
    }
    testID="default-background-image"
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 50,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "space-between",
              "marginBottom": 0,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginTop": 40,
              },
            ]
          }
        >
          <View
            align="center"
            color="grey500"
            fontSize={16}
            lineHeight={24}
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
              ]
            }
          >
            Choose between light and dark mode or set it up automatically, based on your device system settings.
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "marginTop": 32,
                  "paddingHorizontal": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <SvgMock
                height={170}
                width={79}
              />
              <View
                checked={true}
                id={0}
                onChange={[Function]}
                style={
                  Array [
                    Object {
                      "paddingTop": 16,
                    },
                  ]
                }
                testID="0-LIGHT"
                value="LIGHT"
              />
              <View
                align="center"
                fontSize={16}
                lineHeight={24}
              >
                Light
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <SvgMock
                height={170}
                width={79}
              />
              <View
                checked={false}
                id={1}
                onChange={[Function]}
                style={
                  Array [
                    Object {
                      "paddingTop": 16,
                    },
                  ]
                }
                testID="1-DARK"
                value="DARK"
              />
              <View
                align="center"
                fontSize={16}
                lineHeight={24}
              >
                Dark
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <SvgMock
                height={170}
                width={79}
              />
              <View
                checked={false}
                id={2}
                onChange={[Function]}
                style={
                  Array [
                    Object {
                      "paddingTop": 16,
                    },
                  ]
                }
                testID="2-SYSTEM"
                value="SYSTEM"
              />
              <View
                align="center"
                fontSize={16}
                lineHeight={24}
              >
                System
              </View>
            </View>
          </View>
        </View>
        <View
          onPress={[Function]}
          testID="themeChange"
          title="Save"
        />
      </RCTSafeAreaView>
    </View>
  </RCTSafeAreaView>
</View>
`;
